src/classes/THAN_Filter_TDTM.cls

```diff
@@ -61,12 +61,45 @@ public class THAN_Filter_TDTM extends TDTM_Runnable {
 	                    String filterField = String.escapeSingleQuotes(handler.Filter_Field__c);
 	                    String obj = String.escapeSingleQuotes(handler.Object__c);
 	                    if(!String.isBlank(filterField) && !String.isBlank(obj)) {
-		                    try {
-		                        String dynamicQuery = 'select ' + filterField + ' from ' + obj + ' limit 1';
-		                        List<SObject> records =  Database.query(dynamicQuery);
-		                    } catch(QueryException e) {
-		                        handler.addError(Label.InvalidFilter + handler.Class__c);
-		                    }
+                            //those four type of fields are considered valid Filter field
```

Comment:
Some issues here:
1. The relationship field does not work here, need to consider that
2. If we use describe method here, then the object name must have namespace, but it is not true for the existing data
So, might need to reconsider the dynamic query here.

Reply:
1. Fixed relationship validation and logic
2. Fixed the namespace issue for single field in validation 
3. used different validation on relationship and single field